rqt_shell/src/rqt_shell/shell.py

```diff
@@ -85,9 +86,21 @@ def __init__(self, context):
         super(Shell, self).__init__(context)
         self._context = context
         self.setObjectName('Shell')
+        self._args = self._parse_args(context.argv())
 
         self._widget = None
 
+    def _parse_args(self, argv):
+        parser = argparse.ArgumentParser(prog='rqt_shell')
+        Shell.add_arguments(parser)
+        return parser.parse_args(argv)
+
+    @staticmethod
+    def add_arguments(parser):
+        group = parser.add_argument_group('Options for rqt_shell plugin')
+        group.add_argument('-i', '--init_script',
+                            help="load this script after startup in $SHELL")
```

Comment:
Made the help desc better. Added the bit about exporting vars. In this case, executing vs sourcing doesn't make a difference, but still made both the files self-consistent. In either case, the new SHELL doesn't have access to new variables, only access to ```export```.
Another way to run this would be via ```init_file``` but the syntax for each SHELL is different.